src/metabase/pivot/core.cljc

```diff
@@ -556,39 +571,46 @@
                     (let [{:keys [children rawValue isGrandTotal isValueColumn] :as node} (first remaining)
                           path-with-value (if (or isValueColumn isGrandTotal) nil (conj path rawValue))
                           item            (-> (dissoc node :children)
-                                              (assoc :depth depth
-                                                     :offset current-offset
-                                                     :hasChildren (boolean (seq children))
-                                                     :path path-with-value))
-                          item-index      (count result)
-                          _               (conj! result item)
+                                              (assoc :depth depth)
+                                              (assoc :offset current-offset)
+                                              (assoc :hasChildren (boolean (seq children)))
+                                              (assoc :path path-with-value))
+                          item-index      (count @result)
+                          _               (vswap! result conj! item)
                           result-value    (process-tree children (inc depth) current-offset path-with-value)
-                          _               (assoc! result item-index (assoc (nth result item-index)
```

Comment:
@noahmoss Important: you absolutely should reuse the returned values of transient operations, otherwise it can lead to obscure bugs. For example:
```
user=> (let [t (transient {})] (run! #(assoc! t % true) (range 100)) (persistent! t))
{0 true, 1 true, 2 true, 3 true, 4 true, 5 true, 6 true, 7 true}
```

Reply:
Interesting, thanks